Superbuild dependencies in YCM v0.10.1 use build instead of install dir

Tested on [roboticslab-uc3m/asibot-main](https://github.com/roboticslab-uc3m/asibot-main/tree/0259f65fb5c7043319ca0f8d0cc212f15d171b09), a superbuild-like project. Let's focus on the [ROBOTICSLAB_KINEMATICS_DYNAMICS subproject](https://github.com/roboticslab-uc3m/asibot-main/blob/0259f65fb5c7043319ca0f8d0cc212f15d171b09/cmake/ycm-modules/BuildROBOTICSLAB_KINEMATICS_DYNAMICS.cmake), which depends on [COLOR_DEBUG](https://github.com/roboticslab-uc3m/asibot-main/blob/0259f65fb5c7043319ca0f8d0cc212f15d171b09/cmake/ycm-modules/BuildCOLOR_DEBUG.cmake) and [orocos_kdl](https://github.com/roboticslab-uc3m/asibot-main/blob/0259f65fb5c7043319ca0f8d0cc212f15d171b09/cmake/ycm-modules/Buildorocos_kdl.cmake) (among others), and assume we have successfully configured this superbuild project and compiled said dependencies. Next, if we try to build ROBOTICSLAB_KINEMATICS_DYNAMICS, [Travis CI reports an error](https://travis-ci.org/roboticslab-uc3m/asibot-main/jobs/516983168#L9780) due to a missing inclusion:

`/home/travis/build/roboticslab-uc3m/asibot-main/external/ROBOTICSLAB_KINEMATICS_DYNAMICS/libraries/KdlVectorConverterLib/KdlVectorConverter.hpp:8:26: fatal error: kdl/frames.hpp: No such file or directory`

The KdlVectorConverterLib component depends on orocos_kdl (which in turn depends on Eigen3) and COLOR_DEBUG, so let's check the C++ include flags specified in the *flags.make* CMake-generated file at `build/external/ROBOTICSLAB_KINEMATICS_DYNAMICS/libraries/KdlV
ectorConverterLib/CMakeFiles/KdlVectorConverterLib.dir/`:

```
CXX_INCLUDES = \
  -I/root/git/asibot-main/build/external/orocos_kdl/../../../include \
  -I/usr/include/eigen3 \
  -I/root/git/asibot-main/external/ROBOTICSLAB_KINEMATICS_DYNAMICS/libraries/KdlVectorConverterLib \
  -isystem /root/git/asibot-main/external/COLOR_DEBUG 
```

And, now, compare its contents when YCM v0.9.1 is used, instead:

```
CXX_INCLUDES = 
  -I/root/git/asibot-main/build/install/share/orocos_kdl/cmake/../../../include \
  -I/usr/include/eigen3 \
  -I/root/git/asibot-main/external/ROBOTICSLAB_KINEMATICS_DYNAMICS/libraries/KdlVectorConverterLib \
  -isystem /root/git/asibot-main/build/install/include 
```

It seems to me that YCM v0.10.1 is looking for orocos_kdl's and COLOR_DEBUG's include directories in the **build tree** (`root/external/`), whereas YCM v0.9.1 was using the intermediate local **install tree** (`root/build/install/`). This is easier to spot in the COLOR_DEBUG case. In fact, orocos_kdl is known to not be prepared for being consumed by dependent projects without a prior install step (the usual `<...>_DIR` approach), hence its resulting include paths are bogus. Nevertheless, I find it strange that `build/install/` is not used anymore and might be considered a regression.

--- a/help/release/0.10.2.rst
+++ b/help/release/0.10.2.rst
@@ -15,3 +15,4 @@ Generic Modules
 ---------------
 
 * :module:`InstallBasicPackageFiles`: Fixed regular expressions for CMake < 3.9.
+* :module:`YCMEPHelper`: Fixed CMake prefix path of subprojects.

--- a/modules/YCMEPHelper.cmake
+++ b/modules/YCMEPHelper.cmake
@@ -232,9 +232,21 @@ macro(_YCM_SETUP)
   #      directory
   set(_YCM_EP_INSTALL_DIR ${CMAKE_BINARY_DIR}/install)
 
+  # ExternalProject does not handle correctly arguments containing ";" passed
+  # using CMAKE_ARGS, and instead splits them into several arguments. This is
+  # a workaround that replaces ";" with "|" and sets LIST_SEPARATOR "|" in
+  # order to interpret them correctly.
+  #
+  # TODO FIXME check what happens when the "*_COMMAND" arguments are passed.
+  file(TO_CMAKE_PATH "$ENV{CMAKE_PREFIX_PATH}" _CMAKE_PREFIX_PATH)
+  list(INSERT _CMAKE_PREFIX_PATH 0 ${_YCM_EP_INSTALL_DIR})
+  list(REMOVE_DUPLICATES _CMAKE_PREFIX_PATH)
+  string(REPLACE ";" "|" _CMAKE_PREFIX_PATH "${_CMAKE_PREFIX_PATH}")
+  set(_YCM_EP_ALL_CMAKE_ARGS LIST_SEPARATOR "|")
+
   # Default CMAKE_ARGS (Passed to the command line)
   set(_YCM_EP_CMAKE_ARGS "--no-warn-unused-cli"
-                              "-DCMAKE_PREFIX_PATH:PATH=${_CMAKE_PREFIX_PATH}") # Path used by cmake for finding stuff
+                         "-DCMAKE_PREFIX_PATH:PATH=${_CMAKE_PREFIX_PATH}") # Path used by cmake for finding stuff
 
   # Default CMAKE_CACHE_ARGS (Initial cache, forced)
   set(_YCM_EP_CMAKE_CACHE_ARGS "-DCMAKE_INSTALL_PREFIX:PATH=${_YCM_EP_INSTALL_DIR}") # Where to do the installation
@@ -817,18 +829,6 @@ function(YCM_EP_HELPER _name)
                         TMP_DIR ${${_name}_TMP_DIR}
                         STAMP_DIR ${${_name}_STAMP_DIR})
 
-  # ExternalProject does not handle correctly arguments containing ";" passed
-  # using CMAKE_ARGS, and instead splits them into several arguments. This is
-  # a workaround that replaces ";" with "|" and sets LIST_SEPARATOR "|" in
-  # order to interpret them correctly.
-  #
-  # TODO FIXME check what happens when the "*_COMMAND" arguments are passed.
-  file(TO_CMAKE_PATH "$ENV{CMAKE_PREFIX_PATH}" _CMAKE_PREFIX_PATH)
-  list(INSERT _CMAKE_PREFIX_PATH 0 ${${_name}_INSTALL_DIR})
-  list(REMOVE_DUPLICATES _CMAKE_PREFIX_PATH)
-  string(REPLACE ";" "|" _CMAKE_PREFIX_PATH "${_CMAKE_PREFIX_PATH}")
-  set(${_name}_ALL_CMAKE_ARGS LIST_SEPARATOR "|")
-
   # CMAKE_ARGS (Passed to the command line)
   set(${_name}_CMAKE_ARGS CMAKE_ARGS ${_YCM_EP_CMAKE_ARGS})
   if(_YH_${_name}_CMAKE_ARGS)
@@ -845,6 +845,12 @@ function(YCM_EP_HELPER _name)
   set(${_name}_CMAKE_CACHE_DEFAULT_ARGS CMAKE_CACHE_DEFAULT_ARGS ${_YCM_EP_CMAKE_CACHE_DEFAULT_ARGS})
   if(_YH_${_name}_CMAKE_CACHE_DEFAULT_ARGS)
     list(APPEND ${_name}_CMAKE_CACHE_DEFAULT_ARGS ${_YH_${_name}_CMAKE_CACHE_DEFAULT_ARGS})
+  endif()
+
+  # ALL_CMAKE_ARGS
+  set(${_name}_ALL_CMAKE_ARGS ${_YCM_EP_ALL_CMAKE_ARGS})
+  if(_YH_${_name}_ALL_CMAKE_ARGS)
+    list(APPEND ${_name}_ALL_CMAKE_ARGS ${_YH_${_name}_ALL_CMAKE_ARGS})
   endif()
 
   list(APPEND ${_name}_ALL_CMAKE_ARGS ${${_name}_CMAKE_ARGS}

